tests/unit_tests/c_wrappers/job_runner/test_event_reporter.py

```diff
@@ -153,3 +156,57 @@ def test_report_inconsistent_events(unused_tcp_port):
             match=r"Illegal transition None -> \(MessageType<Finish>,\)",
         ):
             reporter.report(Finish())
+
+
+def test_report_with_failed_reporter_but_finished_jobs(unused_tcp_port):
+    mock_send_retry_time = 2
+
+    def mock_send(msg):
+        time.sleep(mock_send_retry_time)
```

Comment:
Yes, it will hang due to calling `join()`, since the thread has finished during the processing of events. Just in case I added a test if the thread is still alive.